doc/encryption_key_rotation.md

```diff
@@ -0,0 +1,86 @@
+# CCDB encryption key rotation
+
+**IMPORTANT** - Always backup the database before rotating the encryption key.
+
+The key used to encrypt the database is generated the first time kubecf is deployed.
+It is based on the Helm values:
+
+```yaml
+ccdb:
+  encryption:
+    rotation:
+      key_labels:
+      - encryption_key_0
+      current_key_label: encryption_key_0
+```
+
+For each label under `key_labels`, kubecf will generate an encryption key.
+The `current_key_label` indicates which key is currently being used.
+
+In order to rotate the CCDB encryption key, add a new label to `key_labels` (keeping the old
+labels), and mark the `current_key_label` with the newly added label. Example:
+
+```yaml
+ccdb:
+  encryption:
+    rotation:
+      key_labels:
+      - encryption_key_0
+      - encryption_key_1
+      current_key_label: encryption_key_1
+```
+
+**IMPORTANT** - key labels should be less than 240 characters long.
+
+Then, update the kubecf Helm installation. After Helm finishes its updates, trigger the
+`rotate-cc-database-key` errand:
+
+**Note** - the following command assumes the Helm installation is named `kubecf` and it was
+installed to the `kubecf` namespace. These values may be different depending on how kubecf was
+installed.
+
+```sh
+kubectl patch qjob rotate-cc-database-key \
+  --namespace kubecf \
+  --type merge \
+  --patch '{"spec":{"trigger":{"strategy":"now"}}}'
+```
+
+## Importing encryption keys
+
+When you import a CCDB database (e.g. via `mysqldump`), then the corresponding encryption
+keys must be imported as well so that the data can be decrypted by the cloud controller.
+
+If the exported data has never had its encryption key rotated, then the only thing to set is
+the top level (legacy) encryption key:
+
+```yaml
+credentials:
+  cc_db_encryption_key: "initial-encryption-key"
+```
+
+After the data has been rotated, all the key labels and values need to be set like this:
+
+```yaml
+ccdb:
+  encryption:
+    rotation:
+      key_labels:
+      - NEW_KEY
+      current_key_label: NEW_KEY
+
+credentials:
+  cc_db_encryption_key: "initial-encryption-key"
+  ccdb_key_label_new_key: "new-encryption-key"
+```
+
+The `key_labels` must be defined **exactly** as they were set in the exporting installation.
```

Comment:
I did the minimal change to say "The imported key_labels ...".

It should be possible (but meaningless unless you make it the new default key) to define an additional label that is not being used by the imported data. That would just be another key rotation though, and should be followed by running the rotation errand. It would be in no way different from doing a rotation after the import at a later time. I don't see why this should be mentioned here; it just complicates the instructions.